lib/improver/utilities/cube_metadata.py

```diff
@@ -279,59 +287,196 @@ def update_attribute(cube, attribute_name, changes, warnings_on=False):
     return result
 
 
+def update_cell_methods(cube, cell_method_definition):
+    """Update cell methods. An "action" keyword is expected within the
+    cell method definition to specify whether the cell method is to be added
+    or deleted. If the cell method defined in the cell_method_definition
+    matches a cell method that already exists within the cube, then the
+    cell method is removed. If the cell method defined in the cell method
+    definition does not match a cell method within the cube, then the
+    cell method is added.
+
+    The cube will be modified in-place.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube containing cell methods that will be updated.
+        cell_method_definition (dict):
+            Dictionary potentially containing the keys: "comments", "coords",
+            "intervals" and "method", as supported by iris.coords.CellMethod.
+            The dictionary must specify an "action" keyword.
```

Comment:
I don't know what you mean by "Any other keys are ignored". The other keys are optional, but are used to build the cell method if they are supplied. It could do with a bit of expansion on the form of the dictionary, just to say that it includes the action key, and then the "method" key as used b the iris.coords.CellMethod function and optionally any of the other keywords supported by iris.coords.CellMethod

Reply:
Reading through the code, the only dictionary keys used from cell_method_definition are "coords", "intervals", "comments", "method", "action". The dictionary is not passed on as **kwargs, so this list should be exhaustive.
My original comment is incorrectly edited from an earlier version. I think we should state the "must specify" keys, "optional keys" and that everything else is ignored without warning although I would prefer to raise a warning to help people who can't speil.